cc-wrapper, bintools-wrapper: Remove now unneeded `set {+,-}u`

Ericson2314 opened the pull request:
###### Motivation for this change

I noticed late in https://github.com/NixOS/nixpkgs/pull/72347 that I missed a few in the wrappers. This could cause more breakage since the `set +u` now "leaks" with out the stdenv cleanup, so I figured rather than last-minute add to the last PR, I'd make another.

https://hydra.nixos.org/jobset/nixpkgs/bash-no-undef-vars

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

Ericson2314 committed:
cc-wrapper, bintools-wrapper: Remove now unneeded `set {+,-}u`

Ericson2314 committed:
cmake: CMAKE_*_PATH can be undefined

Ericson2314 committed:
i-score: CMAKE_PREFIX_PATH can be undefined

Ericson2314 commented on pkgs/development/tools/build-managers/cmake/setup-hook.sh:
```diff
@@ -143,7 +143,7 @@ makeCmakeFindLibs(){
           ;;
         -F*)
-          export CMAKE_FRAMEWORK_PATH="$CMAKE_FRAMEWORK_PATH${CMAKE_FRAMEWORK_PATH:+:}${flag:2}"
+          export CMAKE_FRAMEWORK_PATH="${CMAKE_FRAMEWORK_PATH-}${CMAKE_FRAMEWORK_PATH:+:}${flag:2}"
```
Oops! Good cathc

Ericson2314 committed:
glib: Fix setup hook

It was not robust against multiple matching directories, as the called
function takes exactly 3 arguments.

Ericson2314 committed:
cmake: NIX_CFLAGS_COMPILE NIX_LDFLAG could be undefined

Ericson2314 committed:
opensp: `sgmlHookDone` can be undefined

Ericson2314 committed:
OCaml findlib: OCAMLPATH could be undefined

Ericson2314 committed:
lispPackages.clwrapper: NIX_LISP_LD_LIBRARY_PATH might be undefined

Ericson2314 committed:
lua-5 setup hook: LUA_PATH and LUA_CPATH can be undefined

Ericson2314 committed:
gdk-pixbuf setup hook: GDK_PIXBUF_MODULE_FILE might not be defined

Also switch to [[ ... ]] because it is more robust w.r.t. empty
arguments.

Ericson2314 committed:
darwin private frameworks setup hook: NIX_CFLAGS_COMPILE may be undefined

Ericson2314 committed:
SDL, SDL2: SDL{,2}_PATH may be undefined

Ericson2314 committed:
gstreamer: GST_PLUGIN_SYSTEM_PATH.* may be undefined

Ericson2314 committed:
findlib setup hook: `createFindlibDestdir` may not be defined

Ericson2314 committed:
apple-sdk setup hooks: NIX_CFLAGS_COMPILE may be undefined

jtojnar commented on pkgs/development/libraries/gdk-pixbuf/setup-hook.sh:
```diff
@@ -2,9 +2,9 @@ findGdkPixbufLoaders() {
-			if [ $(cat "$loadersCache"|wc -l) -gt $(cat "$GDK_PIXBUF_MODULE_FILE"|wc -l) ]; then
+	if [[ -f "$loadersCache" ]]; then
+		if [[ -f "${GDK_PIXBUF_MODULE_FILE-}" ]]; then
+			if [[ "$(cat "$loadersCache" | wc -l)" < "$(cat "$GDK_PIXBUF_MODULE_FILE" | wc -l)" ]]; then
```
Should not this rather be `>`

jtojnar commented on pkgs/development/libraries/glib/setup-hook.sh:
```diff
@@ -1,9 +1,11 @@
-    # For packages that need gschemas of other packages (e.g. empathy)
-    if [ -d "$1"/share/gsettings-schemas/*/glib-2.0/schemas ]; then
-        addToSearchPath GSETTINGS_SCHEMAS_PATH "$1/share/gsettings-schemas/"*
+  # For packages that need gschemas of other packages (e.g. empathy)
```
Are not we supposed to use 4 space indent in shell scripts?

Ericson2314 replied:
Are we?

jtojnar replied:
According to [this](https://nixos.org/nixpkgs/manual/#sec-syntax), yes.

Ericson2314 replied:
Oh, thanks.

Ericson2314 committed:
emacs package setup hook: Fix

1. Make the test more robust

2. EMACSLOADPATH may be initially undefined.

3. did `targetOffset` twice when meant `hostOffset` too

Ericson2314 committed:
qt5 qmake setup hook: `qmakeFlags` may be undefined

Ericson2314 commented:
OK, basically all the Darwin errors are due to unreated python and swift problems, and I just fixes the last linux errors I see, so I think this is ready for staging. I'll just watch staging if any darwin problems get uncovered.

matthewbauer commented:
Yeah that sounds good

Ericson2314 committed:
Merge commit 'afa48f16f265fd3e88073bca7929e1e103bd3dc3' into bash-no-undef-vars
